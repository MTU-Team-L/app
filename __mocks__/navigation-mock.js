import {jest} from 'jest-without-globals';

export default {
  setOptions: jest.fn(),
  navigate: jest.fn(),
  isFocused: jest.fn()
};
