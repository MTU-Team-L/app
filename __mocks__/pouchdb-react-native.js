import {jest} from 'jest-without-globals';
import seeds from './seed-data.json';

export const mockPouchDB = jest.fn();

export const put = jest.fn();

const mock = mockPouchDB.mockImplementation(db => {
  return {
    allDocs: jest.fn().mockImplementation(() => Promise.resolve(seeds[db])),
    put,
    changes: jest.fn().mockImplementation(() => ({on: jest.fn()}))
  };
});

export default mock;
