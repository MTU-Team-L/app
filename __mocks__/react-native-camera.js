import React from 'react';
import {View} from 'react-native';

const sampleResponse = [{bounds: {origin: {y: 127.37640380859375, x: 79.48877716064453}, size: {height: 14.498126983642578, width: 80.42393493652344}}, type: 'block', value: 'Jace Beleren', components: [{components: [{bounds: {origin: {y: 127.37640380859375, x: 79.48877716064453}, size: {height: 14.498126983642578, width: 34.60099792480469}}, value: 'Jace', type: 'element'}, {bounds: {origin: {y: 130.48313903808594, x: 119.70074462890625}, size: {height: 9.32022476196289, width: 40.21196746826172}}, value: 'Beleren', type: 'element'}], value: 'Jace Beleren', type: 'line', bounds: {size: {width: 80.42393493652344, height: 14.498126983642578}, origin: {x: 79.48877716064453, y: 127.37640380859375}}}]}, {bounds: {origin: {y: 311.7097473144531, x: 101.93266296386719}, size: {height: 8.28464412689209, width: 54.23939895629883}}, type: 'block', value: 'Planeswalker', components: [{components: [{bounds: {origin: {y: 311.7097473144531, x: 101.93266296386719}, size: {height: 8.28464412689209, width: 54.23939895629883}}, value: 'Planeswalker', type: 'element'}], value: 'Planeswalker', type: 'line', bounds: {size: {width: 54.23939895629883, height: 8.28464412689209}, origin: {x: 101.93266296386719, y: 311.7097473144531}}}]}, {bounds: {origin: {y: 312.7453308105469, x: 171.13465881347656}, size: {height: 8.28464412689209, width: 16.832916259765625}}, type: 'block', value: 'Jace', components: [{components: [{bounds: {origin: {y: 312.7453308105469, x: 171.13465881347656}, size: {height: 8.28464412689209, width: 16.832916259765625}}, value: 'Jace', type: 'element'}], value: 'Jace', type: 'line', bounds: {size: {width: 16.832916259765625, height: 8.28464412689209}, origin: {x: 171.13465881347656, y: 312.7453308105469}}}]}, {bounds: {origin: {y: 332.4213562011719, x: 119.70074462890625}, size: {height: 13.462546348571777, width: 90.71072387695312}}, type: 'block', value: 'ach player drawy& card', components: [{components: [{bounds: {origin: {y: 335.528076171875, x: 119.70074462890625}, size: {height: 9.32022476196289, width: 17.76807975769043}}, value: 'ach', type: 'element'}, {bounds: {origin: {y: 333.4569396972656, x: 139.33914184570312}, size: {height: 11.391386032104492, width: 23.379051208496094}}, value: 'player', type: 'element'}, {bounds: {origin: {y: 333.4569396972656, x: 163.6533660888672}, size: {height: 11.391386032104492, width: 29.925186157226562}}, value: 'drawy&', type: 'element'}, {bounds: {origin: {y: 332.4213562011719, x: 194.51370239257812}, size: {height: 12.426965713500977, width: 15.897754669189453}}, value: 'card', type: 'element'}], value: 'ach player drawy& card', type: 'line', bounds: {size: {width: 90.71072387695312, height: 13.462546348571777}, origin: {x: 119.70074462890625, y: 332.4213562011719}}}]}, {bounds: {origin: {y: 363.48876953125, x: 92.58104705810547}, size: {height: 9.32022476196289, width: 124.37655639648438}}, type: 'block', value: 'CITarget player draws a cand.', components: [{components: [{bounds: {origin: {y: 363.48876953125, x: 92.58104705810547}, size: {height: 9.32022476196289, width: 49.56359100341797}}, value: 'CITarget', type: 'element'}, {bounds: {origin: {y: 365.5599365234375, x: 145.88528442382812}, size: {height: 7.249063491821289, width: 20.573566436767578}}, value: 'player', type: 'element'}, {bounds: {origin: {y: 365.5599365234375, x: 170.19949340820312}, size: {height: 5.177902698516846, width: 19.638402938842773}}, value: 'draws', type: 'element'}, {bounds: {origin: {y: 363.48876953125, x: 191.70822143554688}, size: {height: 9.32022476196289, width: 6.5461344718933105}}, value: 'a', type: 'element'}, {bounds: {origin: {y: 363.48876953125, x: 200.1246795654297}, size: {height: 9.32022476196289, width: 16.832916259765625}}, value: 'cand.', type: 'element'}], value: 'CITarget player draws a cand.', type: 'line', bounds: {size: {width: 124.37655639648438, height: 9.32022476196289}, origin: {x: 92.58104705810547, y: 363.48876953125}}}]}, {bounds: {origin: {y: 380.05804443359375, x: 120.63591003417969}, size: {height: 10.355805397033691, width: 141.20947265625}}, type: 'block', value: 'Target player puts the top wenty cards', components: [{components: [{bounds: {origin: {y: 382.12921142578125, x: 120.63591003417969}, size: {height: 8.28464412689209, width: 22.443889617919922}}, value: 'Target', type: 'element'}, {bounds: {origin: {y: 381.0936279296875, x: 146.82044982910156}, size: {height: 9.32022476196289, width: 20.573566436767578}}, value: 'player', type: 'element'}, {bounds: {origin: {y: 383.164794921875, x: 170.19949340820312}, size: {height: 6.213482856750488, width: 14.962593078613281}}, value: 'puts', type: 'element'}, {bounds: {origin: {y: 381.0936279296875, x: 188.90274047851562}, size: {height: 7.249063491821289, width: 10.286783218383789}}, value: 'the', type: 'element'}, {bounds: {origin: {y: 383.164794921875, x: 201.99501037597656}, size: {height: 6.213482856750488, width: 12.157106399536133}}, value: 'top', type: 'element'}, {bounds: {origin: {y: 382.12921142578125, x: 216.9575958251953}, size: {height: 7.249063491821289, width: 23.379051208496094}}, value: 'wenty', type: 'element'}, {bounds: {origin: {y: 381.0936279296875, x: 243.1421356201172}, size: {height: 7.249063491821289, width: 18.7032413482666}}, value: 'cards', type: 'element'}], value: 'Target player puts the top wenty cards', type: 'line', bounds: {size: {width: 141.20947265625, height: 10.355805397033691}, origin: {x: 120.63591003417969, y: 380.05804443359375}}}]}, {bounds: {origin: {y: 390.4138488769531, x: 93.51620483398438}, size: {height: 8.28464412689209, width: 149.6259307861328}}, type: 'block', value: 'O:of his or her library into his or her', components: [{components: [{bounds: {origin: {y: 390.4138488769531, x: 93.51620483398438}, size: {height: 8.28464412689209, width: 34.60099792480469}}, value: 'O:of', type: 'element'}, {bounds: {origin: {y: 390.4138488769531, x: 130.9226837158203}, size: {height: 7.249063491821289, width: 8.416458129882812}}, value: 'his', type: 'element'}, {bounds: {origin: {y: 390.4138488769531, x: 143.07980346679688}, size: {height: 7.249063491821289, width: 6.5461344718933105}}, value: 'or', type: 'element'}, {bounds: {origin: {y: 390.4138488769531, x: 153.3665771484375}, size: {height: 7.249063491821289, width: 10.286783218383789}}, value: 'her', type: 'element'}, {bounds: {origin: {y: 390.4138488769531, x: 167.39401245117188}, size: {height: 8.28464412689209, width: 22.443889617919922}}, value: 'library', type: 'element'}, {bounds: {origin: {y: 392.4850158691406, x: 193.57855224609375}, size: {height: 4.142322063446045, width: 13.092268943786621}}, value: 'into', type: 'element'}, {bounds: {origin: {y: 390.4138488769531, x: 209.47630310058594}, size: {height: 6.213482856750488, width: 9.3516206741333}}, value: 'his', type: 'element'}, {bounds: {origin: {y: 392.4850158691406, x: 222.56857299804688}, size: {height: 4.142322063446045, width: 6.5461344718933105}}, value: 'or', type: 'element'}, {bounds: {origin: {y: 391.4494323730469, x: 232.85536193847656}, size: {height: 5.177902698516846, width: 10.286783218383789}}, value: 'her', type: 'element'}], value: 'O:of his or her library into his or her', type: 'line', bounds: {size: {width: 149.6259307861328, height: 8.28464412689209}, origin: {x: 93.51620483398438, y: 390.4138488769531}}}]}, {bounds: {origin: {y: 400.7696533203125, x: 117.8304214477539}, size: {height: 7.249063491821289, width: 37.4064826965332}}, type: 'block', value: 'graveyard.', components: [{components: [{bounds: {origin: {y: 400.7696533203125, x: 117.8304214477539}, size: {height: 7.249063491821289, width: 37.4064826965332}}, value: 'graveyard.', type: 'element'}], value: 'graveyard.', type: 'line', bounds: {size: {width: 37.4064826965332, height: 7.249063491821289}, origin: {x: 117.8304214477539, y: 400.7696533203125}}}]}, {bounds: {origin: {y: 412.1610412597656, x: 160.84786987304688}, size: {height: 10.355805397033691, width: 50.49875259399414}}, type: 'block', value: 'Ak Bril', components: [{components: [{bounds: {origin: {y: 414.2322082519531, x: 160.84786987304688}, size: {height: 8.28464412689209, width: 25.24937629699707}}, value: 'Ak', type: 'element'}, {bounds: {origin: {y: 412.1610412597656, x: 193.57855224609375}, size: {height: 7.249063491821289, width: 17.76807975769043}}, value: 'Bril', type: 'element'}], value: 'Ak Bril', type: 'line', bounds: {size: {width: 50.49875259399414, height: 10.355805397033691}, origin: {x: 160.84786987304688, y: 412.1610412597656}}}]}];

// eslint-disable-next-line import/no-mutable-exports
let triggerTextRecognition = () => {};

const RNCamera = ({onTextRecognized}) => {
  triggerTextRecognition = async () => {
    await onTextRecognized({textBlocks: sampleResponse});
  };

  return <View/>;
};

RNCamera.Constants = {
  Type: {
    back: 'mock-camera-back'
  },
  FlashMode: {
    on: 'mock-on'
  }
};

export {RNCamera, triggerTextRecognition};
