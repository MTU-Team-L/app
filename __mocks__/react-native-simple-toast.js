import {jest} from 'jest-without-globals';

const mock = {
  SHORT: 'short',
  LONG: 'long',
  show: jest.fn()
};

module.exports = mock;
