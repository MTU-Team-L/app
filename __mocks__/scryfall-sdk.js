import {jest} from 'jest-without-globals';
import {scryfallCard} from './seed-data.json';

const Cards = {
  byName: () => Promise.resolve(scryfallCard),
  search: jest.fn(),
  autoCompleteName: jest.fn()
};

export {Cards};
