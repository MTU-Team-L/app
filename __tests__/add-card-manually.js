import React from 'react';
import {test, expect} from 'jest-without-globals';
import {render, fireEvent} from 'react-native-testing-library';
import AddCardManually from '../src/scenes/add-card-manually';
import {put} from '../__mocks__/pouchdb-react-native';
import navigationMock from '../__mocks__/navigation-mock';

test('renders correctly', async () => {
  const {toJSON} = await render(<AddCardManually navigation={navigationMock}/>);

  expect(toJSON()).toMatchSnapshot();
});

test('can add card', async () => {
  const {getByText, getByPlaceholder} = await render(<AddCardManually navigation={navigationMock}/>);

  // Set up mock
  put.mockReturnValueOnce(Promise.resolve({id: 'Test-ID', ok: true, rev: '1-1ac49f882c114c68904c14c73095f484'}));

  await fireEvent.changeText(getByPlaceholder('Card name'), 'Disperse');

  await fireEvent.press(getByText('Add'));

  expect(put.mock.calls[0]).toMatchSnapshot();
});
