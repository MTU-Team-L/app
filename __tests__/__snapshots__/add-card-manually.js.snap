// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can add card 1`] = `
Array [
  Object {
    "_id": "Disperse",
    "arena_id": 67780,
    "artist": "Ryan Yee",
    "artist_ids": Array [
      "8955dca7-3e37-42b4-83a9-167c78a2178f",
    ],
    "booster": true,
    "border_color": "black",
    "card_back_id": "0aeebaf5-8c7d-4636-9e82-8c27447861f7",
    "cmc": 2,
    "collector_number": "50",
    "color_identity": Array [
      "U",
    ],
    "colors": Array [
      "U",
    ],
    "digital": false,
    "edhrec_rank": 5219,
    "flavor_text": "It's pointless to hold on when you have nothing to hold on with.",
    "foil": true,
    "frame": "2015",
    "full_art": false,
    "games": Array [
      "arena",
      "mtgo",
      "paper",
    ],
    "highres_image": true,
    "id": "2ff7c89c-41dc-4ac3-bbce-38ab86f435ea",
    "illustration_id": "74dbe86f-c808-41db-8e6e-e59ca3d73aea",
    "image_uris": Object {
      "art_crop": "https://img.scryfall.com/cards/art_crop/front/2/f/2ff7c89c-41dc-4ac3-bbce-38ab86f435ea.jpg?1562301229",
      "border_crop": "https://img.scryfall.com/cards/border_crop/front/2/f/2ff7c89c-41dc-4ac3-bbce-38ab86f435ea.jpg?1562301229",
      "large": "https://img.scryfall.com/cards/large/front/2/f/2ff7c89c-41dc-4ac3-bbce-38ab86f435ea.jpg?1562301229",
      "normal": "https://img.scryfall.com/cards/normal/front/2/f/2ff7c89c-41dc-4ac3-bbce-38ab86f435ea.jpg?1562301229",
      "png": "https://img.scryfall.com/cards/png/front/2/f/2ff7c89c-41dc-4ac3-bbce-38ab86f435ea.png?1562301229",
      "small": "https://img.scryfall.com/cards/small/front/2/f/2ff7c89c-41dc-4ac3-bbce-38ab86f435ea.jpg?1562301229",
    },
    "lang": "en",
    "layout": "normal",
    "legalities": Object {
      "brawl": "not_legal",
      "commander": "legal",
      "duel": "legal",
      "future": "not_legal",
      "historic": "legal",
      "legacy": "legal",
      "modern": "legal",
      "oldschool": "not_legal",
      "pauper": "legal",
      "penny": "legal",
      "pioneer": "legal",
      "standard": "not_legal",
      "vintage": "legal",
    },
    "mana_cost": "{1}{U}",
    "mtgo_id": 68261,
    "multiverse_ids": Array [
      447186,
    ],
    "name": "Disperse",
    "nonfoil": true,
    "object": "card",
    "oracle_id": "9d8fe3f8-5027-4f9e-999e-e277caa96478",
    "oracle_text": "Return target nonland permanent to its owner's hand.",
    "oversized": false,
    "prices": Object {
      "eur": "0.04",
      "tix": "0.04",
      "usd": "0.04",
      "usd_foil": "0.17",
    },
    "prints_search_uri": "https://api.scryfall.com/cards/search?order=released&q=oracleid%3A9d8fe3f8-5027-4f9e-999e-e277caa96478&unique=prints",
    "promo": false,
    "purchase_uris": Object {
      "cardhoarder": "https://www.cardhoarder.com/cards/68261?affiliate_id=scryfall&ref=card-profile&utm_campaign=affiliate&utm_medium=card&utm_source=scryfall",
      "cardmarket": "https://www.cardmarket.com/en/Magic/Products/Singles/Core-2019/Disperse?referrer=scryfall&utm_campaign=card_prices&utm_medium=text&utm_source=scryfall",
      "tcgplayer": "https://shop.tcgplayer.com/product/productsearch?id=166703&partner=Scryfall&utm_campaign=affiliate&utm_medium=api&utm_source=scryfall",
    },
    "rarity": "common",
    "related_uris": Object {
      "edhrec": "https://edhrec.com/route/?cc=Disperse",
      "gatherer": "https://gatherer.wizards.com/Pages/Card/Details.aspx?multiverseid=447186",
      "mtgtop8": "https://mtgtop8.com/search?MD_check=1&SB_check=1&cards=Disperse",
      "tcgplayer_decks": "https://decks.tcgplayer.com/magic/deck/search?contains=Disperse&page=1&partner=Scryfall&utm_campaign=affiliate&utm_medium=api&utm_source=scryfall",
    },
    "released_at": "2018-07-13",
    "reprint": true,
    "reserved": false,
    "rulings_uri": "https://api.scryfall.com/cards/2ff7c89c-41dc-4ac3-bbce-38ab86f435ea/rulings",
    "scryfall_set_uri": "https://scryfall.com/sets/m19?utm_source=api",
    "scryfall_uri": "https://scryfall.com/card/m19/50/disperse?utm_source=api",
    "set": "m19",
    "set_name": "Core Set 2019",
    "set_search_uri": "https://api.scryfall.com/cards/search?order=set&q=e%3Am19&unique=prints",
    "set_type": "core",
    "set_uri": "https://api.scryfall.com/sets/2f5f2509-56db-414d-9a7e-6e312ec3760c",
    "story_spotlight": false,
    "tcgplayer_id": 166703,
    "textless": false,
    "type_line": "Instant",
    "uri": "https://api.scryfall.com/cards/2ff7c89c-41dc-4ac3-bbce-38ab86f435ea",
    "variation": false,
  },
]
`;

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 0.8,
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          placeholder="Card name"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Add
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
